common/persistence/sql/sqlplugin/sqlite/typeconv.go

```diff
@@ -0,0 +1,49 @@
+// The MIT License
+//
+// Copyright (c) 2021 Datadog, Inc.
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sqlite
+
+import "time"
+
+type (
+	// DataConverter defines the API for conversions to/from
+	// go types to sqlite datatypes
+	DataConverter interface {
+		ToSQLiteDateTime(t time.Time) time.Time
+		FromSQLiteDateTime(t time.Time) time.Time
+	}
+	converter struct{}
+)
+
+// ToSQLiteDateTime converts to time to SQLite datetime
+func (c *converter) ToSQLiteDateTime(t time.Time) time.Time {
```

Comment:
plz make sure sqlite time conversion does not have limit

postgresql reference https://github.com/temporalio/temporal/blob/master/common/persistence/sql/sqlplugin/postgresql/typeconv.go